src/runner/api/itemRunner.js

```diff
@@ -434,11 +424,71 @@ var itemRunnerFactory = function itemRunnerFactory(providerName, data, options)
          *      renderingQueue; // {'feedback2'}
          *    });
          */
-        renderFeedbacks: function renderFeedbacks(feedbacks, itemSession, done) {
-            if (_.isFunction(provider.renderFeedbacks)) {
+        renderFeedbacks(feedbacks, itemSession, done) {
+            if (typeof provider.renderFeedbacks === 'function') {
                 provider.renderFeedbacks.call(this, feedbacks, itemSession, done);
             }
-        }
+        },
+
+        /**
+         * Call the provider's suspend method
+         * @returns {Promise}
+         */
+        suspend() {
+            if (!suspended && !closed && flow.render.done  && typeof provider.suspend === 'function') {
+                return provider.suspend.call(this).then(result => {
+                    suspended = true;
+                    return result;
+                });
+            }
+            return Promise.resolve();
+        },
+
+        /**
+         * Call the provider's hide method
+         * @returns {Promise}
+         */
+        close() {
+            if (!closed && flow.render.done && typeof provider.close === 'function') {
+                return provider.close.call(this).then(result => {
+                    closed = true;
+                    return result;
+                });
+            }
+            return Promise.resolve();
+        },
+
+        /**
+         * Call the provider's resume method.
+         * We can resume a previously suspended or closed item.
+         * @returns {Promise}
+         */
+        resume() {
+            if ( (suspended || closed) && flow.render.done && typeof provider.resume === 'function') {
```

Comment:
According to the flow diagram, a `closed` itemSession can only transition to `review` or `solution`. Doesn't it mean that we should only be allowed to resume a `suspended` session?

Reply:
yes, in the QTI implementation. I kept the generic implementation open.